pkg/kubelet/cm/memorymanager/policy_static_test.go

```diff
@@ -3816,11 +3834,135 @@ func TestStaticPolicyGetTopologyHints(t *testing.T) {
+					MemoryMap: map[v1.ResourceName]*state.MemoryTable{
+						v1.ResourceMemory: {
+							Allocatable:    2 * gb,
+							Free:           2 * gb,
+							Reserved:       0,
+							SystemReserved: 512 * mb,
+							TotalMemSize:   3 * gb,
+						},
+						hugepages1Gi: {
+							Allocatable:    gb,
+							Free:           gb,
+							Reserved:       0,
+							SystemReserved: 0,
+							TotalMemSize:   gb,
+						},
+					},
+					Cells: []int{0},
+				},
+				1: &state.NUMANodeState{
+					MemoryMap: map[v1.ResourceName]*state.MemoryTable{
+						v1.ResourceMemory: {
+							Allocatable:    2 * gb,
+							Free:           2 * gb,
+							Reserved:       0,
+							SystemReserved: 512 * mb,
+							TotalMemSize:   3 * gb,
+						},
+						hugepages1Gi: {
+							Allocatable:    gb,
+							Free:           gb,
+							Reserved:       0,
+							SystemReserved: 0,
+							TotalMemSize:   gb,
+						},
+					},
+					Cells: []int{1},
+				},
+			},
+			systemReserved: systemReservedMemory{
+				0: map[v1.ResourceName]uint64{
+					v1.ResourceMemory: 1024 * mb,
+				},
+				1: map[v1.ResourceName]uint64{
+					v1.ResourceMemory: 1024 * mb,
+				},
+			},
+			machineInfo: &cadvisorapi.MachineInfo{
+				Topology: []cadvisorapi.Node{
+					{
+						Id:     0,
+						Memory: 3 * gb,
+						HugePages: []cadvisorapi.HugePagesInfo{
+							{
+								PageSize: pageSize1Gb,
+								NumPages: 1,
+							},
+						},
+					},
+					{
+						Id:     1,
+						Memory: 3 * gb,
+						HugePages: []cadvisorapi.HugePagesInfo{
+							{
+								PageSize: pageSize1Gb,
+								NumPages: 1,
+							},
+						},
+					},
+				},
+			},
+			expectedTopologyHints: map[string][]topologymanager.TopologyHint{
+				string(v1.ResourceMemory): {
+					{
+						NUMANodeAffinity: newNUMAAffinity(0),
+						Preferred:        true,
+					},
+					{
+						NUMANodeAffinity: newNUMAAffinity(1),
+						Preferred:        true,
+					},
+					{
+						NUMANodeAffinity: newNUMAAffinity(0, 1),
+						Preferred:        false,
+					},
+				},
+				string(hugepages1Gi): {
+					{
+						NUMANodeAffinity: newNUMAAffinity(0),
+						Preferred:        true,
+					},
+					{
+						NUMANodeAffinity: newNUMAAffinity(1),
+						Preferred:        true,
+					},
+					{
+						NUMANodeAffinity: newNUMAAffinity(0, 1),
+						Preferred:        false,
+					},
+				},
+			},
+			podLevelResourcesEnabled:        true,
+			podLevelResourceManagersEnabled: true,
+		}, {
+			description: "should not provide pod topology hints for burstable pod with pod level resources when PodLevelResourceManagers is enabled",
+			pod:         getPodWithPodLevelResources("pod1", podLevelRequirementsBurstable, "container1", requirementsGuaranteed),
+			systemReserved: systemReservedMemory{
+				0: map[v1.ResourceName]uint64{
+					v1.ResourceMemory: 1024 * mb,
+				},
+			},
+			expectedTopologyHints:           nil,
+			podLevelResourcesEnabled:        true,
+			podLevelResourceManagersEnabled: true,
+		},
```

Comment:
it would be nice to have full coverage of cases: pod level resources which result in guaranteed QoS which override container-level resources (= generate different hints) and sanity check with both pod level and container level resources which lead to burstable QoS. Nice to have for alpha.

Reply:
Added more combinations to cover those cases for both CPU and Memory managers.